### PR TITLE
Mysql fixes - sqlparse is not mandatory; conditions for "ALL" grant check for mysql 8.0

### PR DESCRIPTION
**Pull request by bp2work**
### What issues does this PR fix or reference?
Fixes: salt/module/mysql.py

### Previous Behavior
1) lib sqlparse was mandatory (module:re was replaced by module:sqlparse)
2) ALL PRIVILEGES were transformed to a list of all available privileges even NOT for GLOBAL context (`database = '*.*'`)

### New Behavior
1) lib sqlparse is not mandatory (use module:sqlparse if available, otherwise use previous realization with module:re)
2) ALL PRIVILEGES transforms to a list ONLY for GLOBAL context (`database = '*.*'`) check

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by bp2work** [comment]
I can not see, what is the problem with jenkins scripts.
Failed linters are not critical (or even meaningless, imho), you can fix/skip them.

**Comment by garethgreenaway** [comment]
@bp2work Thanks for the PR.  In this case the lint and pre-commit failures are legitimate.  Your changes to the mysql test introduced an extra newline at the end of the file and you need to install & run the pre-commit pieces to your PR.  These run a variety of checks and updates against the files that you've changed.  More information about pre-commit can be found here: https://docs.saltproject.io/en/latest/topics/development/contributing.html#quickstart

**Comment by dwoz** [comment]
> I can not see, what is the problem with jenkins scripts.
> Failed linters are not critical (or even meaningless, imho), you can fix/skip them.

@bp2work I understand your sentiment here. Linters often seem like more of a pain than something helpful. However, in Salt's case, we have a codebase with more than 30 million lines of code. Making sure the automated tests pass and the code conforms to specific format requirements helps tremendously in the effort to maintain the project. We very much appreciate you working within our strict processes to make your contribution.

File: salt/modules/mysql.py [hunk 1]
```diff
@@ -2374,7 +2395,11 @@ def grant_exists(
         if (
             salt.utils.versions.version_cmp(server_version, "8.0") >= 0
             and "MariaDB" not in server_version
+            and database == "*.*"
```
Comment (bp2work) [inline salt/modules/mysql.py:2398]: if my tests are broken, just add this ONE line to any patch and reject my PR

**Comment by bp2work** [comment]
1. One empty line removed, linter checks are successful.
2. [Pre-commit checks](https://github.com/saltstack/salt/pull/59724/checks?check_run_id=2068431085) fail on `black` module, on 2-3 strings. And they should be **like now given**, it's more readable. (module `black` is in beta test now).
3. Some checks here (ci) failed with message "This check failed", without any information (I think, it failed by timeout):
-  [Twine Check #1326](https://github.com/saltstack/salt/pull/59724/checks?check_run_id=2068431313)
-  [Lint #2774](https://github.com/saltstack/salt/pull/59724/checks?check_run_id=2068431338)

4. Some jenkins tests failed (?), I can not test them.

P.S.
I recommend to merge with 'no-ff' option, to see each commit (from feature branch) in main branch.
And about tests: I understand all requirements to this big project. And I see a HUGE number of commits without ANY tests at all.
I'm not a tester, I'm not even a developer. I lost 2 days to try to run these checks on my PC. I don't want anymore have any deals with virtualenvs and a huge pack of foreign utils (pytest, nox, pre-commit, etc). I'll just write my own patches and use them.
https://github.com/saltstack/salt/pull/59724/files#r590583380 - this is critical, decide

**Comment by bp2work** [comment]
> Just following up here @bp2work . You are no longer working on this PR correct?

Hi! Just close it.
1. I'm not working on it already
2. Critical fixes were already merged from another branches
3. Turning back `libsqlparse` should be in another PR

**Pull request closed without merging**